app/models/workflow.rb

```diff
@@ -29,16 +30,54 @@ def table
   end
 
   def new_sequence
-    return move_rest_higher if sequence
+    throw :abort if sequence && sequence <= 0
 
-    self.sequence = self.class.last&.sequence.to_i + 1
+    largest = last_sequence
+    if sequence && sequence < last_sequence
+      sequence_higher(sequence)
+    else
+      self.sequence = largest + 1 # auto_assignment if sequence is nil or too large
+    end
   end
 
-  # move all related sequence number one higher if the desired number is in use
-  def move_rest_higher
-    return unless sequence && sequence_changed? && self.class.exists?(:sequence => sequence)
+  # no gap between sequences
+  def adjust_sequences
+    return unless sequence_changed?
 
-    self.class.where(table[:sequence].gteq(sequence)).update_all("sequence = (-sequence - 1)")
-    self.class.where(table[:sequence].lt(0)).update_all("sequence = (-sequence)")
+    throw :abort if sequence && sequence <= 0
+
+    largest = last_sequence
+    self.sequence = largest if sequence.nil? || sequence > largest
+
+    if sequence > sequence_was
+      sequence_lower(sequence_was, sequence)
+    else
+      sequence_higher(sequence, sequence_was)
+    end
+  end
+
+  # sequences increment between [sequence sequence_was sequence)
+  def sequence_higher(startp, endp = nil)
+    query = self.class.where(table[:sequence].gteq(startp))
+    query = query.where(table[:sequence].lteq(endp)) if endp
+    query.update_all("sequence = (-sequence - 1)")
```

Comment:
Do you think it makes sense to extract this block out (as well as 72-74) into a separate method that has a good name of what this is doing and pass in the `-1` from 63. And then for 72-74 it could also use this new method, but pass in `1` instead. Small "DRY".